src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp

```diff
@@ -189,7 +204,56 @@ struct ONNXLoopOpLowering : public ConversionPattern {
           thenBlock.end(), loopBodyBlock.getOperations());
       rewriter.eraseBlock(&loopBodyBlock);
     }
-    rewriter.replaceOp(op, outputs);
+
+    rewriter.restoreInsertionPoint(afterLoop);
+    // accumulate dynamic tensor
+    // Convert the memref<?xmemref<>> to a memref
+    SmallVector<Value, 4> newOutputs;
+    for (auto output : outputs) {
+      auto seqElementType =
+          output.getType().cast<MemRefType>().getElementType();
+      if (seqElementType.isa<MemRefType>()) {
+        // need convertion
+        // ToDo: need a IF statement to handle output is empty
+        // we can safely give 0 to the dynamic dim for alloc
+        // Here loop is assumed to be executed at least once.
+        auto firstElement =
+            create.krnl.load(output, create.math.constantIndex(0));
+        SmallVector<mlir::Value, 4> allocParams;
+        SmallVector<int64_t, 4> dims;
+        dims.emplace_back(output.getType().cast<MemRefType>().getShape()[0]);
+        if (output.getType().cast<MemRefType>().getShape()[0] == -1)
```

Comment:
How about `if (dims[0] == -1)`?

Reply:
If you are asking dims[0] for the `output`, the dynamic value is given when the memref is allocated and we can use dim op to get it.
If you are asking dims[0] of the firstElement, the dynamic value can be retrieved with dim op when the memref is loaded from memref<*xmemref<>>. The dynamic dim size is part of the runtime data structure.